pkg/kubelet/kubelet.go

```diff
@@ -913,7 +923,7 @@ func (kl *Kubelet) GenerateRunContainerOptions(pod *api.Pod, container *api.Cont
 			opts.PodContainerDir = p
 		}
 	}
-	if pod.Spec.DNSPolicy == api.DNSClusterFirst {
+	if pod.Spec.DNSPolicy == api.DNSClusterFirst || kl.resolverConfig != ResolvConfDefault {
```

Comment:
In fact, if you want to use this resolv.conf instead of inheriting from the host, you want to process it even if DNSPolicy is not using cluster DNS